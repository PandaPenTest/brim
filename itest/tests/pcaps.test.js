/* @noflow */
import {readdirSync, unlinkSync} from "fs"
import path from "path"

import {LOG} from "../lib/log"
import {
  click,
  clickPcapButton,
  newAppInstance,
  pcapIngestSample,
  pcapsDir,
  searchDisplay,
  startApp,
  startSearch,
  waitForSearch,
  waitUntilDownloadFinished,
  writeSearch
} from "../lib/app"
import {selectors} from "../../src/js/test/integration"
import {handleError, stdTest} from "../lib/jest"

const clearPcaps = async (app) => {
  let dir = await pcapsDir(app)
  let files = readdirSync(dir)
  files.forEach((fileBasename) => {
    if (fileBasename.match(/^packets-.+\.pcap$/)) {
      let fileAbspath = path.join(dir, fileBasename)
      unlinkSync(fileAbspath)
      LOG.debug(`Unlinked file ${fileAbspath}`)
    }
  })
}

describe("Test PCAPs", () => {
  let app
  let testIdx = 0

  beforeEach(async () => {
    app = newAppInstance(path.basename(__filename), ++testIdx)
    await startApp(app)
    await clearPcaps(app)
    await pcapIngestSample(app)
  })

  afterEach(async () => {
    await clearPcaps(app)
    if (app && app.isRunning()) {
      return await app.stop()
    }
  })

  stdTest(
    "pcap button downloads deterministically-formed pcap file",
    (done) => {
      writeSearch(
        app,
        "_path=ssl id.orig_h=192.168.1.110 id.resp_h=209.216.230.240 id.resp_p=443"
      )
        .then(async () => {
          await startSearch(app)
          await waitForSearch(app)
          await searchDisplay(app)
          await click(app, selectors.viewer.resultCellContaining("ssl"))
          await clickPcapButton(app)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    }
  )
})
