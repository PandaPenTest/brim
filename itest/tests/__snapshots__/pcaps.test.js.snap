// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PCAPs pcap button downloads deterministically-formed pcap file 1`] = `
Array [
  "ts",
  "_path",
  "uid",
  "id.orig_h",
  "id.orig_p",
  "id.resp_h",
  "id.resp_p",
  "version",
  "cipher",
  "curve",
  "server_name",
  "resumed",
  "last_alert",
  "next_protocol",
  "established",
  "cert_chain_fuids",
  "client_cert_chain_fuids",
  "subject",
  "issuer",
  "client_subject",
  "client_issuer",
  "validation_status",
  "02/25/2016:0314.0210",
  "ssl",
  "CBdn7C5Fbcmgp64Te",
  "192.168.1.110",
  "57640",
  "209.216.230.240",
  "443",
  "TLSv12",
  "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
  "secp256r1",
  "news.ycombinator.com",
  "F",
  "⦻",
  "http/1.1",
  "T",
  "FXkLzR25cwYgeI5yI7",
  "Frbo5s1CRsV02JUoSe",
  "F1VUER3CattO8PiHWc",
  "CN=news.ycombinator.com,O=Y Combinator\\\\, Inc.,L=San Francisco,ST=California,C=US",
  "CN=DigiCert SHA2 Secure Server CA,O=DigiCert Inc,C=US",
  "⦻",
  "⦻",
  "ok",
]
`;
